Gordon360/Services/LostAndFoundService.cs

```diff
@@ -210,5 +249,17 @@ public IEnumerable<MissingItemReportViewModel> GetMissingItemsAll()
             }
             return (MissingItemReportViewModel)report;
         }
+
+        /// <summary>
+        /// Gets a list of Actions Taken by id
+        /// </summary>
+        /// <param name="id">The ID of the associated missing item report</param>
+        /// <returns>An ActionsTaken, or null if no item matches the id</returns>
+        public IEnumerable<ActionsTakenViewModel> GetActionsTaken(int id)
+        {
+            IEnumerable<ActionsTaken> actionsList = context.ActionsTaken.Where(x => x.missingID == id);
```

Comment:
Currently, this doesn't return the submitter's username to the frontend.  Missing item does this using a SQL view, but in this context that shouldn't be necessary, and we can just get the username using some code here.

When you get the submitter ID from the database (which you have up until you typecast it into the viewModel), you can use submitterID to find the submitterUsername in the account table (basically the reverse of getting the id number based on the username, which you did in the POST request).

Reply:
Added code to pass the username based on the submitter ID